app/components/EditProfileForm.tsx

```diff
@@ -1,17 +1,58 @@
 'use client'
 import { FormEvent } from 'react';
+import { useRouter } from 'next/navigation';
 
-export default function EditProfileForm() {
+export default function EditProfileForm({userData}) {
 
-  /**
-   * Submits new post data
-   * 
-   * @param e - the form submit event
-  */
-  const onSubmitHandler = (e: FormEvent<HTMLFormElement>): void => {
+  const router = useRouter();
+
+  const firstName = userData['data'][0]['firstName'][0]
+  const lastName = userData['data'][0]['lastName'][0]
+  const username = userData['data'][0]['username'][0]
+  const gender = userData['data'][0]['gender'][0]
+  const bio = userData['data'][0]['bio'][0]
+
+  const userInfo = new Map([
+    ['firstName', firstName],
+    ['lastName', lastName],
+    ['username', username],
+    ['gender', gender],
+    ['bio', bio]
+  ])
```

Comment:
Any reason for the Map as opposed to a normal JSON object?

`const userInfo = { 'firstName': firstName, 'lastName': lastName, ... }`

You should still be able to do `userInfo[key] === value` later on in your logic.

Reply:
@datderek You're right, this stackoverflow post details the differences between maps and objects in javascript: https://stackoverflow.com/questions/18541940/map-vs-object-in-javascript 

They are different but for our use case an object will suffice.

Reply:
Sounds good! Yeah, since we don't need to iterate in-order (Map), lets just stick to an Object.